src/Cardano/PoolActions.hs

```diff
@@ -0,0 +1,40 @@
+module Cardano.PoolActions where
+
+import           Cardano.Models
+import           Cardano.Utils
+import           ErgoDex.Contracts.Pool
+import           ErgoDex.Contracts.Types
+import           ErgoDex.OffChain
+import           Plutus.V1.Ledger.Value
+import qualified PlutusTx.AssocMap   as Map
+import           Ledger.Scripts
+import qualified Ledger.Typed.Scripts.Validators as Validators
+import qualified PlutusTx
+import           Data.List
+import           Data.Maybe
+
+createPool :: PoolParams -> FullTxOut -> FullTxOut -> FullTxOut -> TxCandidate
+createPool pp@PoolParams{..} inputWithX inputWithY inputWithNft = let
+  commonValue   = foldl (\acc curOutput -> acc <> (fullTxOutValue curOutput)) (Value Map.empty) [inputWithX, inputWithY, inputWithNft]
+  toInputs      = map fromPkTxOut2TxIn [inputWithX, inputWithY, inputWithNft]
+  newPoolOutput = TxOutCandidate {
+		txOutCandidateAddress = Validators.validatorAddress poolInstance,
+		txOutCandidateValue = commonValue,
+		txOutCandidateDatum = Just $ Datum (PlutusTx.toBuiltinData pp)
+  } in TxCandidate {
+		txCandidateInputs = toInputs,
+		txCandidateOutputs = [newPoolOutput]
+  }
+
+initPool :: PoolParams -> PoolParams -> FullTxOut -> TxCandidate
```

Comment:
Pool params are always the same for a given pool, why old and new? Also, error cases?